Added Netlify tutorial

ajeetraina opened the pull request.

simonprickett commented:
@ajeetraina is this ready for review, I found `http://localhost:3000/develop/guides/netlify/getting-started` looks like a page stub at the moment when I run the site locally at the moment?

wjohnsto commented on docs/explore/netlify/getting-started-with-netlify/index-getting-started-with-netlify.mdx:
```diff
@@ -0,0 +1,305 @@
+Netlify is a popular static site hosting serverless platform. It is a popular way to build, deploy, and scale modern web applications in a much more scalable and secure way.  
+
+
+Netlify helps developers to launch websites and campaigns in minutes with no-ops. The hosting platform is built on top of JAMstack and unifies JavaScript and APIs,strong enough to create an infrastructure that’s well suited for both developers and content editors.
```
with ~no-ops~ no fuss. ~The hosting platform is built on top of JAMstack and unifies JavaScript and APIs,strong enough to create an infrastructure that’s well suited for both developers and content editors.~ Netlify is built primarily for JAMstack sites, which unify JavaScript and APIs to allow applications that are well suited for both developers and content editors.

wjohnsto commented on docs/explore/netlify/getting-started-with-netlify/index-getting-started-with-netlify.mdx:
```diff
@@ -0,0 +1,305 @@
+
+- It delivers out-of-the-box continuous integration.
+- The platform allows continuous deployment through its support for Git repository deployment.
+- It allows developers to focus on building and deploying apps by asbtracting all the maintenance work away from the developers.
```
~asbtracting~ abstracting

wjohnsto commented on docs/explore/netlify/getting-started-with-netlify/index-getting-started-with-netlify.mdx:
```diff
@@ -0,0 +1,305 @@
+
+![MyImages](netlifyimage.png)
+
+1. The developer writes a code and store it in a version control repository(say, GitHub).
```
The developer writes code and stores it in a version control repository (e.g. GitHub).

wjohnsto commented on docs/explore/netlify/getting-started-with-netlify/index-getting-started-with-netlify.mdx:
```diff
@@ -0,0 +1,305 @@
+![MyImages](netlifyimage.png)
+
+1. The developer writes a code and store it in a version control repository(say, GitHub).
+2. When a new change is merged into the master branch of GitHub repo, Github triggers a webhook that Netlify is listening for. 
```
When a new change is merged into the main branch of the repository, a webhook notifies Netlify to deploy a new site.

wjohnsto commented on docs/explore/netlify/getting-started-with-netlify/index-getting-started-with-netlify.mdx:
```diff
@@ -0,0 +1,305 @@
+
+1. The developer writes a code and store it in a version control repository(say, GitHub).
+2. When a new change is merged into the master branch of GitHub repo, Github triggers a webhook that Netlify is listening for. 
+3. Netlify pulls the latest version of the app from GitHub and then start building the static version of the site
```
Netlify pulls the latest version of the app from the repository and runs a build command to generate the static site files

wjohnsto commented on docs/explore/netlify/getting-started-with-netlify/index-getting-started-with-netlify.mdx:
```diff
@@ -0,0 +1,305 @@
+1. The developer writes a code and store it in a version control repository(say, GitHub).
+2. When a new change is merged into the master branch of GitHub repo, Github triggers a webhook that Netlify is listening for. 
+3. Netlify pulls the latest version of the app from GitHub and then start building the static version of the site
+4. Netlify then uses Plugins and internal code to make adjustments to your site, pre-render all of your pages in static HTML and improves it further
```
Remove this step, it is unnecessary and Netlify doesn't do anything specific here.

wjohnsto commented on docs/explore/netlify/getting-started-with-netlify/index-getting-started-with-netlify.mdx:
```diff
@@ -0,0 +1,305 @@
+2. When a new change is merged into the master branch of GitHub repo, Github triggers a webhook that Netlify is listening for. 
+3. Netlify pulls the latest version of the app from GitHub and then start building the static version of the site
+4. Netlify then uses Plugins and internal code to make adjustments to your site, pre-render all of your pages in static HTML and improves it further
+5. Once the build process gets completed, Netlify then pushes the contents of the  directory to its global CDN
```
Once the build process gets completed, Netlify takes the static assets and pushes them to its global CDN for fast delivery

wjohnsto commented on docs/explore/netlify/getting-started-with-netlify/index-getting-started-with-netlify.mdx:
```diff
@@ -0,0 +1,305 @@
+4. Netlify then uses Plugins and internal code to make adjustments to your site, pre-render all of your pages in static HTML and improves it further
+5. Once the build process gets completed, Netlify then pushes the contents of the  directory to its global CDN
+
+Let us see how Netlify deploys a simple React app flawlessly.
```
Let's walk through deploying a simple React app to Netlify.

ajeetraina replied:
Raja suggested to add Caching app rather than this app that has no "Redis" component into it.

wjohnsto commented on sidebars.js:
```diff
@@ -282,7 +282,15 @@ module.exports = {
+           label: 'Netlify',
+           items: [
+            'explore/netlify/getting-started-with-netlify/index-getting-started-with-netlify',
+            'explore/netlify/deploy-redis-developersite-clone/index-deploy-redis-developersite-clone',
```
make sure to adjust this if you adjust the title

wjohnsto reviewed (changes requested):
Here is my feedback, once it's done we can go live!

ajeetraina commented:
Hi @simonprickett @wjohnsto I need to make the below changes:
1. Restructure it and put it under "Create" section.(rather than explore).
2. The getting started guide must talk about Redis in some way, so will change the example a bit.

Will be making the changes and sending you back for the review.

ajeetraina commented on docs/explore/netlify/deploy-redis-developersite-clone/index-deploy-redis-developersite-clone.mdx:
```diff
@@ -0,0 +1,129 @@
+---
+id: index-deploy-redis-developersite-clone
+title: How to Deploy and Host Redis Developer Site Clone with Netlify in 5 Minutes
```
That's a good suggestion. let me make the change.

ajeetraina commented on docs/explore/netlify/deploy-redis-developersite-clone/index-deploy-redis-developersite-clone.mdx:
```diff
@@ -0,0 +1,129 @@
+id: index-deploy-redis-developersite-clone
+title: How to Deploy and Host Redis Developer Site Clone with Netlify in 5 Minutes
+sidebar_label: Deply and Host Redis Developer Site Clone with Netlify in 5 Minutes
+slug: /explore/netlify/deploy-redis-developersite-clone
```
Sure. Will push the change.

ajeetraina commented:
@simonprickett @wjohnsto  I have modified the blog and added the reference to https://github.com/redis-developer/nextjs-redis-netlify GITHUB repo this time. Please have a look and let me know.

simonprickett commented on docs/create/aws/index-aws.mdx:
```diff
@@ -60,8 +60,8 @@ The following links provide you with the available options to run apps on Heroku
      <RedisCard
-        title="How to Build and Deploy Your Own Analytics Dashboard on AWS using NodeJS and Redis"
-        description="Learn how to Build and Deploy an Analytics Dashboard using NodeJS and then deploy it on AWS"
+        title="How to Build and Deploy Your Own Analytics Dashboard using NodeJS and Redis on AWS Platform"
```
```suggestion
        title="How to Build and Deploy Your Own Analytics Dashboard using NodeJS and Redis on the AWS Platform"
```

simonprickett commented on docs/create/netlify/deploy-docusaurus-to-netlify/index-deploy-docusaurus-to-netlify.mdx:
```diff
@@ -0,0 +1,128 @@
+ <div class="col">
+     <RedisCard
+        title="How to get started with Netlify"
+        description="Learn how to deploy and host Next.js app using Netlify "
```
```suggestion
        description="Learn how to deploy and host a Next.js app using Netlify "
```

simonprickett commented on docs/create/netlify/deploy-docusaurus-to-netlify/index-deploy-docusaurus-to-netlify.mdx:
```diff
@@ -0,0 +1,128 @@
+</div>
+
+
+[In the last tutorial](/create/netlify/getting-started-with-netlify), we leveraged [Netlify CLI](https://docs.netlify.com/cli/get-started/) to build a simple Next.js application built using TailwindCSS and Redis .
```
```suggestion
[In the last tutorial](/create/netlify/getting-started-with-netlify), we leveraged [Netlify CLI](https://docs.netlify.com/cli/get-started/) to build a simple Next.js application built using TailwindCSS and Redis.
```

simonprickett commented on docs/create/netlify/deploy-docusaurus-to-netlify/index-deploy-docusaurus-to-netlify.mdx:
```diff
@@ -0,0 +1,128 @@
+
+![MyImage](images/netlify-signin.png)
+
+### Step 2. Connect your Netlify to Git Provider
```
```suggestion
### Step 2. Connect your Netlify account to your Git Provider
```

simonprickett commented on docs/create/netlify/deploy-docusaurus-to-netlify/index-deploy-docusaurus-to-netlify.mdx:
```diff
@@ -0,0 +1,128 @@
+
+### Step 2. Connect your Netlify to Git Provider
+
+Netlify provides you to sign-in using various ways - GitHub, GitLab, Bitbucket, Email and SSO.
```
```suggestion
Netlify allows you to sign-in using various services, including GitHub, GitLab, Bitbucket, Email and SSO.
```

simonprickett commented on docs/create/netlify/deploy-docusaurus-to-netlify/index-deploy-docusaurus-to-netlify.mdx:
```diff
@@ -0,0 +1,128 @@
+
+### Step 5. Configure site settings for Netlify 
+
+There are two essential settings that needs to be configured.
```
```suggestion
There are two essential settings that need to be configured.
```

simonprickett commented on docs/create/netlify/deploy-docusaurus-to-netlify/index-deploy-docusaurus-to-netlify.mdx:
```diff
@@ -0,0 +1,128 @@
+### Step 5. Configure site settings for Netlify 
+
+There are two essential settings that needs to be configured.
+First, change the URL under docusaurus.config.js to any other random URL as shown below:
```
```suggestion
First, change the URL under `docusaurus.config.js` to any other random URL as shown below:
```

simonprickett commented on docs/create/netlify/deploy-docusaurus-to-netlify/index-deploy-docusaurus-to-netlify.mdx:
```diff
@@ -0,0 +1,128 @@
+![MyImage](images/netlify_sitedeploy-inprogress.png)
+
+
+Monitor the "Deploy Log" carefully to see if there is no serious error message appear in the log.
```
```suggestion
Monitor the "Deploy Log" carefully to see if any error messages appear in the log.
```

simonprickett commented on docs/create/netlify/deploy-docusaurus-to-netlify/index-deploy-docusaurus-to-netlify.mdx:
```diff
@@ -0,0 +1,128 @@
+
+![MyImage](images/netlify-deploylog.png)
+
+By now, you should be able to see your Docusaurus site hosted on port 3000.
```
```suggestion
You should now be able to see your Docusaurus site hosted on port 3000.
```

simonprickett commented on docs/create/netlify/deploy-docusaurus-to-netlify/index-deploy-docusaurus-to-netlify.mdx:
```diff
@@ -0,0 +1,128 @@
+
+![MyImage](images/netlify_clickpreview.png)
+
+You will able to see that Netlify upload sites assets to a content delivery network and makes your site available.
```
```suggestion
You will able to see that Netlify uploads site assets to a content delivery network and makes your site available.
```

simonprickett reviewed (approved):
Left some suggestions.
